### DISCUSSION TITLE
How do I set-up my connection string ?

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#15271] [Q&A]
I am watching a EF core sqlite tutorial and the guy is setting the connection string in the appsettings.json file (which is in the project). However, I don't have such file in my project. Any help?

**Comment by timunie** [comment]
Well check out Newtonsoft.Json and System.Text.Json (iirc) packages on how to use Json files. It's not really Avalonia specific. 

> [!NOTE]
> I'd only use EF core if I don't need trimming nor AOT as it is currently not supported by them. So only Desktop makes sense here.

**Comment by thevortexcloud** [comment]
The appsettings.json file is just the default in ASP Core. ASP Core does a lot of stuff for you out of the box, such as set up config file support, Depedency injection, etc. You don't need it though to specify a connection string. If you really want to add a json config file yourself though just add a reference to the `Microsoft.Extensions.Configuration.Json` package and manually point it at your config file (the file can be called anything since you are telling it where to find it) when setting up your config builder. This will give you the same functionality as what ASP Core has for config file support.

The connection string can be passed to your EF context from pretty much anywhere, including your own custom config system (EG passing it as a launch argument to your app). You can even hard code it you really want to (do NOT hard code it if your app is publicly distributed and your connection string contains credentials). 

I should note EF Core is not optimised for desktop apps. It's meant for web servers. You can use it in a desktop app though, but the experience may be below subpar. 

I personally normally just use Dapper, it works pretty well in desktop apps and is significantly simpler than EF Core.
